pyoptsparse/pySNOPT/pySNOPT.py

```diff
@@ -663,12 +663,22 @@ def _snstop(self, ktcond, mjrprtlvl, minimize, n, nncon, nnobj, ns, itn, nmajor,
                 if "funcs" in self.cache.keys():
                     iterDict["funcs"].update(self.cache["funcs"])
 
+        # Create the restart dictionary to be passed to snstop_handle
+        restartDict = {
```

Comment:
TODO for the future: this should probably be a dataclass object instead of a dict... That way we can avoid some code duplication with above.

Reply:
@ewu63  I can do it if you open an issue with the specification of what you would like to have ;-)

Reply:
Opened #405

Reply:
Thanks for the comments. It would be nice for the function to be more flexible. I'm not sure how returning values would work though because the user does not call `_snstop`. It seems like the dictionaries have to be passed in.